### PR TITLE
Azure.Core: Add benchmark for creating and sending a message using HttpPipeline

### LINKED ISSUE #40753: ClientModel: Develop mock tests for performance benchmarking
This is needed as part of the Azure.Core 2.0 work to:
- isolate possible regressions independent of network stack
- do profiling focused on Core library code only

It is a prerequisite to the validation step needed to ensure that performance regressions aren't introduced with Azure.Core moves to take a dependency on System.ClientModel

### PR DESCRIPTION
**Pull request by annelo-msft**
We have validated that integrating Azure.Core with System.ClientModel does not significantly increase the time it takes to create and send an HttpMessage using Azure.Core's HttpPipeline in https://github.com/Azure/azure-sdk-for-net/issues/41033#issuecomment-1905016043.

This PR checks in the benchmark used to do the test so it can be used for later analysis.

The intention is that anyone making changes to pipeline types in either Azure.Core or System.ClientModel will run this benchmark on the main branch and then separately check out their PR branch and run the test again, both on their own machine.  The resulting numbers should be included in the description of the Core PR.

Fixes https://github.com/Azure/azure-sdk-for-net/issues/40753

### CODE REVIEW

File: sdk/core/Azure.Core/perf/HttpPipelineBenchmark.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        ClientOptions options = new BenchmarkOptions
+        {
+            Transport = new HttpClientTransport(new HttpClient(new MockHttpMessageHandler()))
+        };
+
+        _pipeline = HttpPipelineBuilder.Build(options);
+    }
+
+    [Benchmark]
+    public async Task CreateAndSendMessage()
```
Comment (KrzysztofCwalina) [inline sdk/core/Azure.Core/perf/HttpPipelineBenchmark.cs:32]: Could you add the same test for System.ClientModel pipeline?

File: sdk/core/Azure.Core/perf/HttpPipelineBenchmark.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Jobs;
+
+namespace Azure.Core.Perf;
+
+[SimpleJob(RuntimeMoniker.Net60)]
```
Comment (KrzysztofCwalina) [inline sdk/core/Azure.Core/perf/HttpPipelineBenchmark.cs:15]: Could you add an allocation anayzer to show how much each test allocates?

Comment (annelo-msft) [reply sdk/core/Azure.Core/perf/HttpPipelineBenchmark.cs:15]: I had originally added this, but then saw that it was added automatically in the [benchmark runner's config](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/perf/Program.cs#L33).  I'm fine adding it explicitly to the class definition if it has no effect and someone wanted to run it outside this program file implementation.

Comment (KrzysztofCwalina) [reply sdk/core/Azure.Core/perf/HttpPipelineBenchmark.cs:15]: Ah, I did not realize we have that in the config.

Comment (annelo-msft) [reply sdk/core/Azure.Core/perf/HttpPipelineBenchmark.cs:15]: I am serious that I literally typed it in and then went and checked and deleted it.  I am happy to add it back so we don't make this mistake in the future - we have both spent more time on it now than we needed to :)

Let me see if adding it as an attribute too has any negative side-effect.

Comment (annelo-msft) [reply sdk/core/Azure.Core/perf/HttpPipelineBenchmark.cs:15]: Looks like there's no problem with this; will add

File: sdk/core/Azure.Core/perf/HttpPipelineBenchmark.cs (hunk 1 above)
Comment (annelo-msft) [reply sdk/core/Azure.Core/perf/HttpPipelineBenchmark.cs:33]: Added

Comment (KrzysztofCwalina) [reply sdk/core/Azure.Core/perf/HttpPipelineBenchmark.cs:33]: Should the S.CM model test be in the same test class so we could compare S.CM to Azure.Core?

Comment (annelo-msft) [reply sdk/core/Azure.Core/perf/HttpPipelineBenchmark.cs:33]: Good suggestion

**Review by KrzysztofCwalina — APPROVED** [review]

**Pull request merged**